In [ ]:
! git clone https://github.com/JK8789/C-AdvDGM.git

fatal: destination path 'C-AdvDGM' already exists and is not an empty directory.


In [1]:
!python --version


Python 3.9.20


In [ ]:
#!pip install -q condacolab
#import condacolab
#condacolab.install()

⏬ Downloading https://github.com/conda-forge/miniforge/releases/download/23.11.0-0/Mambaforge-23.11.0-0-Linux-x86_64.sh...
📦 Installing...
📌 Adjusting configuration...
🩹 Patching environment...
⏲ Done in 0:00:15
🔁 Restarting kernel...


In [27]:
!pip uninstall conda

Found existing installation: conda 23.11.0
Uninstalling conda-23.11.0:
  Would remove:
    /usr/local/bin/conda
    /usr/local/lib/python3.10/site-packages/conda-23.11.0.dist-info/*
    /usr/local/lib/python3.10/site-packages/conda/*
    /usr/local/lib/python3.10/site-packages/conda_env/*
    /usr/local/lib/python3.10/site-packages/tests/conda_env/*
    /usr/local/lib/python3.10/site-packages/tests/test-recipes/activate_deactivate_package/src/etc/conda/activate.d/activate.bat
    /usr/local/lib/python3.10/site-packages/tests/test-recipes/activate_deactivate_package/src/etc/conda/activate.d/activate.sh
    /usr/local/lib/python3.10/site-packages/tests/test-recipes/activate_deactivate_package/src/etc/conda/deactivate.d/deactivate.bat
    /usr/local/lib/python3.10/site-packages/tests/test-recipes/activate_deactivate_package/src/etc/conda/deactivate.d/deactivate.sh
Proceed (Y/n)? Y
  Successfully uninstalled conda-23.11.0


In [ ]:
#!conda --version


conda 23.11.0


In [ ]:
#!conda info -e

# conda environments:
#
base                     /usr/local
project                  /usr/local/envs/project



In [ ]:
#!conda create -n "project" python=3.9

Channels:
 - conda-forge
Platform: linux-64
Solving environment: \ | done


==> WARNING: A newer version of conda exists. <==
    current version: 23.11.0
    latest version: 24.9.2

Please update conda by running

    $ conda update -n base -c conda-forge conda



## Package Plan ##

  environment location: /usr/local/envs/project

  added / updated specs:
    - python=3.9


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bzip2-1.0.8                |       h4bc722e_7         247 KB  conda-forge
    ca-certificates-2024.8.30  |       hbcca054_0         155 KB  conda-forge
    ld_impl_linux-64-2.43      |       h712a8e2_2         654 KB  conda-forge
    libgcc-14.2.0              |       h77fa898_1         829 KB  conda-forge
    libgcc-ng-14.2.0           |       h69a702a_1          53 KB  conda-forge
    libgomp-14.2.0             |       h77fa898_1         450 KB  conda-forge
    l

In [ ]:
import condacolab
condacolab.install() # expect a kernel restart

✨🍰✨ Everything looks OK!


In [6]:
#!apt install python3.9

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libpython3.9-minimal libpython3.9-stdlib mailcap mime-support python3.9-minimal
Suggested packages:
  python3.9-venv binfmt-support
The following NEW packages will be installed:
  libpython3.9-minimal libpython3.9-stdlib mailcap mime-support python3.9 python3.9-minimal
0 upgraded, 6 newly installed, 0 to remove and 49 not upgraded.
Need to get 5,284 kB of archives.
After this operation, 19.4 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 mailcap all 3.70+nmu1ubuntu1 [23.8 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 mime-support all 3.66 [3,696 B]
Get:3 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 libpython3.9-minimal amd64 3.9.20-1+jammy1 [837 kB]
Get:4 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 python3.9-minimal am

In [28]:
!apt remove python3.9

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libpython3.9-minimal libpython3.9-stdlib python3.9-minimal
Use 'apt autoremove' to remove them.
The following packages will be REMOVED:
  python3.9
0 upgraded, 0 newly installed, 1 to remove and 49 not upgraded.
After this operation, 586 kB disk space will be freed.
(Reading database ... 124288 files and directories currently installed.)
Removing python3.9 (3.9.20-1+jammy1) ...
Processing triggers for mailcap (3.70+nmu1ubuntu1) ...
Processing triggers for man-db (2.10.2-1) ...


In [24]:
!update-alternatives --install /usr/local/bin/python python /usr/bin/python3.9 3

update-alternatives: renaming python link from /usr/bin/python to /usr/local/bin/python


In [10]:
!file /etc/alternatives/python

/etc/alternatives/python: symbolic link to /usr/bin/python3.9


In [5]:
!python --version

Python 3.9.20


In [3]:
#!eval "$(conda shell.bash hook)" # copy conda command to shell
#!sudo ln -s /opt/conda/root/etc/profile.d/conda.sh /etc/profile.d/conda.s
#!eval "$(/usr/local/condabin/conda shell.bash hook)"
#!cat ~/.bashrc
#!conda init
#!conda activate myenv
#!python --version

ln: failed to create symbolic link '/etc/profile.d/conda.s': File exists
# If not running interactively, don't do anything
[ -z "$PS1" ] && return

# don't put duplicate lines in the history. See bash(1) for more options
# ... or force ignoredups and ignorespace
HISTCONTROL=ignoredups:ignorespace

# append to the history file, don't overwrite it
shopt -s histappend

# for setting history length see HISTSIZE and HISTFILESIZE in bash(1)
HISTSIZE=1000
HISTFILESIZE=2000

# check the window size after each command and, if necessary,
# update the values of LINES and COLUMNS.
shopt -s checkwinsize

# make less more friendly for non-text input files, see lesspipe(1)
[ -x /usr/bin/lesspipe ] && eval "$(SHELL=/bin/sh lesspipe)"

PS1='\[\033\]\w\[\033\]\$ '

# enable color support of ls and also add handy aliases
if [ -x /usr/bin/dircolors ]; then
    test -r ~/.dircolors && eval "$(dircolors -b ~/.dircolors)" || eval "$(dircolors -b)"
    alias ls='ls --color=auto'
    #alias dir='dir --color=au

In [2]:
!conda init

no change     /usr/local/condabin/conda
no change     /usr/local/bin/conda
no change     /usr/local/bin/conda-env
no change     /usr/local/bin/activate
no change     /usr/local/bin/deactivate
no change     /usr/local/etc/profile.d/conda.sh
no change     /usr/local/etc/fish/conf.d/conda.fish
no change     /usr/local/shell/condabin/Conda.psm1
no change     /usr/local/shell/condabin/conda-hook.ps1
no change     /usr/local/lib/python3.10/site-packages/xontrib/conda.xsh
no change     /usr/local/etc/profile.d/conda.csh
no change     /root/.bashrc
No action taken.


In [3]:
#!conda activate base


CondaError: Run 'conda init' before 'conda activate'



In [17]:
!python --version

Python 3.9.20


In [6]:
import os
os.getcwd()
os.chdir("/content/C-AdvDGM")
os.getcwd()
#!pwd
#!ls


'/content/C-AdvDGM'

In [44]:
!pip install -r requirements.txt

In [6]:
import mlc.run.train_search

# Hyperparameter search

In [10]:
mlc.run.train_search.run(dataset_name="lcld_201317_ds_time")

Output hidden; open in https://colab.research.google.com to view.

Results:  model_name: tabtransformer.
dataset_name="lcld_201317_ds_time".

Best parameters: {'x_metadata': None, 'objective': 'classification', 'use_gpu': True, 'batch_size': 256, 'num_classes': 2, 'early_stopping_rounds': 5, 'val_batch_size': 1024, 'class_weight': 'balanced', 'custom_dataloader': '', 'scaler_type': 'TabScaler', 'depth': 1, 'heads': 2, 'weight_decay': -1, 'learning_rate': -3, 'dropout': 0.3, 'data_parallel': False, 'epochs': 100, 'model_name': 'tabtransformer', 'dataset': 'lcld_201317_ds_time', 'num_splits': 5, 'seed': 42, 'shuffle': True, 'metrics': ['auc'], 'dim': 32, 'task': 'HP train', 'trial': 'best', 'timestamp': '2024-11-10 11:44:18'} with best performance: 0.6291545759743103


In [11]:
mlc.run.train_search.run(dataset_name="lcld_201317_ds_time", save_best_model=1)

Train hyperparameter optimization for tabtransformer on lcld_201317_ds_time


[I 2024-11-10 16:03:53,467] Using an existing study with name 'tabtransformer_lcld_201317_ds_time_TabScaler' instead of creating a new one.


100/100 already completed


/usr/local/lib/python3.10/site-packages/sklearn/preprocessing/_encoders.py:868: FutureWarning: `sparse` was renamed to `sparse_output` in version 1.2 and will be removed in 1.4. `sparse_output` is ignored unless you leave `sparse` to its default value.
  warnings.warn(
COMET WARNING: display_summary is set in Comet config but has been deprecated; use display_summary_level instead
COMET WARNING: To get all data logged automatically, import comet_ml before the following modules: sklearn, torch.


KeyboardInterrupt: 

In [12]:
mlc.run.train_search.run(dataset_name="url", save_best_model=1)

Output hidden; open in https://colab.research.google.com to view.

Results:
model_name: tabtransformer, dataset_name="url"

Best parameters: {'x_metadata': None, 'objective': 'classification', 'use_gpu': True, 'batch_size': 256, 'num_classes': 2, 'early_stopping_rounds': 5, 'val_batch_size': 1024, 'class_weight': 'balanced', 'custom_dataloader': '', 'scaler_type': 'TabScaler', 'depth': 2, 'heads': 4, 'weight_decay': -3, 'learning_rate': -3, 'dropout': 0.2, 'data_parallel': False, 'epochs': 100, 'model_name': 'tabtransformer', 'dataset': 'url', 'num_splits': 5, 'seed': 42, 'shuffle': True, 'metrics': ['auc'], 'dim': 64, 'task': 'HP train', 'trial': 'best', 'timestamp': '2024-11-10 18:02:54'} with

best performance: 0.9711234493797519

Best trial parameters: {'depth': 2, 'dim': 64, 'dropout': 0.2, 'heads': 4, 'learning_rate': -3, 'weight_decay': -3} with best performance: 0.9711234493797519
/usr/local/lib/python3.10/site-packages/sklearn/preprocessing/_encoders.py:868: FutureWarning: `sparse` was renamed to `sparse_output` in version 1.2 and will be removed in 1.4. `sparse_output` is ignored unless you leave `sparse` to its default value.
  warnings.warn(


Test accuracy: 0.9243219597550306

Test f1: 0.9248152976966536

Test f1_weighted: 0.9243187012481505

Test precision: 0.918825561312608

Test recall: 0.9308836395450568

Test auc: 0.9675192066433516


In [23]:
!pip install pandas

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.9/89.9 kB 2.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.1/13.1 MB 41.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 346.6/346.6 kB 11.1 MB/s eta 0:00:00


In [17]:
import pandas as pd
help(pd)

Output hidden; open in https://colab.research.google.com to view.

In [29]:
!python --version

Python 3.10.12


In [26]:
!apt remove miniconda

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package miniconda


In [12]:
!sudo update-alternatives --config python

There are 2 choices for the alternative python (providing /usr/local/bin/python).

  Selection    Path                 Priority   Status
------------------------------------------------------------
* 0            /usr/bin/python3.9    3         auto mode
  1            /usr/bin/python3.10   1         manual mode
  2            /usr/bin/python3.9    3         manual mode

Press <enter> to keep the current choice[*], or type selection number: 1
update-alternatives: using /usr/bin/python3.10 to provide /usr/local/bin/python (python) in manual mode


In [33]:
!pip install optuna

  Using cached optuna-4.0.0-py3-none-any.whl.metadata (16 kB)
Using cached optuna-4.0.0-py3-none-any.whl (362 kB)


In [8]:
!python --version

Python 3.9.20


In [34]:
import sys
print(sys.version)
#!pip install optuna
!/content/C-AdvDGM/bash_scripts/hyperparam_search/wgan.sh

3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]
Varying the learning rate for 
Traceback (most recent call last):
  File "/content/C-AdvDGM/run_advdgm/main_wgan.py", line 7, in <module>
    import optuna
ModuleNotFoundError: No module named 'optuna'
Traceback (most recent call last):
  File "/content/C-AdvDGM/run_advdgm/main_wgan.py", line 7, in <module>
    import optuna
ModuleNotFoundError: No module named 'optuna'
Traceback (most recent call last):
  File "/content/C-AdvDGM/run_advdgm/main_wgan.py", line 7, in <module>
    import optuna
ModuleNotFoundError: No module named 'optuna'
Traceback (most recent call last):
  File "/content/C-AdvDGM/run_advdgm/main_wgan.py", line 7, in <module>
    import optuna
ModuleNotFoundError: No module named 'optuna'
Traceback (most recent call last):
  File "/content/C-AdvDGM/run_advdgm/main_wgan.py", line 7, in <module>
    import optuna
ModuleNotFoundError: No module named 'optuna'
python: can't open file '/content/C-AdvDGM/run_dgm/main_wgan.py